knimin/lib/data_access.py

```diff
@@ -2906,18 +3043,19 @@ def edit_sample_plate(self, id, name, email=None, created_on=None,
             Defines the plate type of the sample plate
             If None, the first plate type will be used
         """
-        self._sample_plate_exists(id)
-        self._sample_plate_is_unique(name, id)
-        if email:
-            self._email_exists(email)
-        plate_type_id = self._plate_type_name_to_id(plate_type)
         with TRN:
+            self._sample_plate_exists(id)
```

Comment:
I'm just gonna mention it here although there is no necessity to change at this moment. Usually is a bad idea to use variable names that match functions defined in python. `id` is a function in python so it will probably be a better idea to name the variable `plate_id`. In this function, you no longer have access to the python function `id`, and although you're not using it, it is usually a better practice not to overwrite it. This is just FYI, and no need to change now, although it will be nice to eventually change it.